tests/Database/DatabaseEloquentBuilderTest.php

```diff
@@ -526,6 +526,21 @@ public function testHasWithContraintsAndHavingInSubquery()
         $this->assertEquals(['baz', 'qux', 'quuux'], $builder->getBindings());
     }
 
+    public function testHasWithContraintsWithOrWhereAndHavingInSubquery()
+    {
+        $model = new EloquentBuilderTestModelParentStub;
+
+        $builder = $model->where('bar', 'baz');
+        $builder->whereHas('foo', function ($q) {
+            $q->having('bam', '>', 'qux');
+            $q->where('baam', '>', 'quux');
+            $q->orWhere('baaam', '>', 'quuux');
+        })->where('quux', 'quuuux');
+
```

Comment:
@wleona3 Can you maybe use some more "human readable" example? like users, posts, etc. instead of quuux and baaam...